repos/fdbt-site/src/pages/api/pointToPointPeriodProduct.ts

```diff
@@ -0,0 +1,66 @@
+import { NextApiResponse } from 'next';
+import { updateSessionAttribute } from '../../utils/sessions';
+import { ErrorInfo, NextApiRequestWithSession, PointToPointPeriodProduct } from '../../interfaces';
+import { redirectTo, redirectToError } from './apiUtils';
+import { POINT_TO_POINT_PRODUCT_ATTRIBUTE } from '../../constants/attributes';
+import {
+    checkDurationIsValid,
+    checkProductNameIsValid,
+    isValidInputDuration,
+    removeExcessWhiteSpace,
+} from './apiUtils/validator';
+
+export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
+    try {
+        const { productNameInput, productDuration, durationUnits } = req.body;
+        const whiteSpaceCleansedNameInput = removeExcessWhiteSpace(productNameInput);
+        const whiteSpaceCleansedDurationInput = removeExcessWhiteSpace(productDuration);
+        const pointToPointPeriodProduct: PointToPointPeriodProduct = {
+            productName: whiteSpaceCleansedNameInput,
+            productDuration: whiteSpaceCleansedDurationInput,
+            productDurationUnits: durationUnits,
+        };
+
+        const nameCheckError = checkProductNameIsValid(whiteSpaceCleansedNameInput);
+        const productDurationUnitsCheckError = !isValidInputDuration(durationUnits, true)
+            ? 'Select a valid expiry unit'
+            : '';
+        let durationCheckError = '';
+        if (durationUnits !== 'no expiry') {
+            durationCheckError = checkDurationIsValid(whiteSpaceCleansedDurationInput);
+        }
+
+        const errors: ErrorInfo[] = [];
+
+        if (nameCheckError || durationCheckError || productDurationUnitsCheckError) {
```

Comment:
Minor: I think it would be better to build the errors array as we go rather than having to have a long list in an if statement - it's borderline with 3 error but it would make it more maintainable for the future

Reply:
Pushed a commit to change this, @jacsamell is this better?

Reply:
Perfect 👍

Reply:
this one